modules/hal_silabs/simplicity_sdk/src/sl_memory_manager_pool_shim.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2025 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Shim basic sl_* pool functions to Zephyr k_mem_slab functions,
+ * which again will get redirected to the Zephyr sys_heap.
+ */
+
+#include <stdlib.h>
+
+#include <zephyr/kernel.h>
+
+#include "sl_memory_manager.h"
+#include "sl_status.h"
+
+sl_status_t sl_memory_create_pool(size_t block_size, uint32_t block_count,
+				  sl_memory_pool_t *pool_handle)
+{
+	struct k_mem_slab *slab = k_malloc(sizeof(struct k_mem_slab));
+
+	char *slab_buffer = k_aligned_alloc(4, block_size * block_count);
```

Comment:
This changes the heap you're using to something else than what we already use e.g. for `sl_malloc()`. I was probably too unclear in my earlier comment - sorry about that. I made a reference to the system heap just to show that some separate API may be needed to get aligned pointers. If we stick to the libc heap (which we probably should) you need some other solution for this.